v2/authorisation/middleware.go

```diff
@@ -74,12 +95,50 @@ func (m PermissionCheckMiddleware) Require(permission string, handlerFunc http.H
 
 		authToken = strings.TrimPrefix(authToken, "Bearer ")
 
-		entityData, err := m.jwtParser.Parse(authToken)
-		if err != nil {
-			logData["message"] = err.Error()
-			log.Info(ctx, "authorisation failed due to jwt parsing issue", logData)
-			w.WriteHeader(http.StatusForbidden)
-			return
+		var (
+			chunks = strings.Split(authToken, ".")
+			headerData = tokenHeaderData{}
+		)
+		// is the token of the form xxx.yyy.zzz (i.e. JWT)?
+		if len(chunks) == chunkSize {
+			sDec, decodeErr := b64.StdEncoding.DecodeString(chunks[0])
+			if decodeErr != nil {	
+				log.Error(ctx, "authorisation failed due to issue decoding authorisation token", decodeErr, logData)
+				w.WriteHeader(http.StatusForbidden)
+				return
+			}
```

Comment:
LGTM. Would this check be sufficient to confirm if this is a JWT ?

Reply:
Thanks @pidah - its step 1 of a 2 step check; the second step attempts to check the `typ` field in the `headerData` struct against a constant:

https://github.com/ONSdigital/dp-authorisation/blob/a6f65bf395704a4b3b7684761315e3d0316878a2/v2/authorisation/middleware.go#L124

Reply:
oh my bad didn't see that... LGTM ! 🙂